### ISSUE TITLE
Compilation Error for example mutualEx

### ISSUE THREAD

**Issue by weltoph** [paraVerifier/paraVerifier#2]
Upon following the steps provided in the README I obtain the following error message after executing
`corebuild mutualEx.byte -pkg re2 -I src`:
```
+ ocamlfind ocamldep -package re2 -package core -ppx 'ppx-jane -as-ppx' -modules src/paramecium.mli > src/paramecium.mli.depends
File "src/paramecium.mli", line 13, characters 0-4:
Error: Syntax error
Command exited with code 2.
Compilation unsuccessful after building 4 targets (3 cached) in 00:00:00.
```
The Syntax error refers to the `with` in (taken from `src/paramecium.mli`):
```
(** Constants *)
type const =
  | Intc of int
  | Strc of string
  | Boolc of bool
with sexp
```
Am I doing something wrong or did I potentially miss a step somewhere?

**Comment by weltoph** [comment]
Initial research suggests that we have a mismatch in the ocaml-versions. Could you be so kind to specify which ocaml-version was used here? I operate on `The OCaml toplevel, version 4.07.1`.

**Comment by weltoph** [comment]
Okay, I think I got a working setup now. For anyone else trying to re-create this, these are the steps I took:
Starting from an Ubuntu 14.04.5 LTS, I installed OCaml (`The OCaml toplevel, version 4.01.0`). Then I built opam (`1.2.2`) from source and installed the packages in the README.md (note here that utop needs a more recent OCaml version and therefore needs to be pinned to something ...not so recent, for me this is utop1.19, this also changes installation order because cohttp requires a version of lwt which is not compatible with utop 1.19. Hence: first get utop and install cohttp afterwards!). Additionally, NuSMV needs to be in version 2.5.4 (the current version 2.6.0 does _not_ work). Additionally, the generated scripts for Isabelle cannot be interpreted with Isabelle2017 nor Isabelle2018. I used Isabelle2014 which seem to work. But now everything seems to work fine.
